src/Microsoft.ApiDesignGuidelines.Analyzers/Core/NonConstantFieldsShouldNotBeVisible.cs

```diff
@@ -24,15 +25,25 @@ public abstract class NonConstantFieldsShouldNotBeVisibleAnalyzer : DiagnosticAn
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182353.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterSymbolAction(obj =>
```

Comment:
Given how similar these rules are I think it's better if both diagnostics are issued by one analyzer so that it doesnt lead to confusion about missing cases. We could just have this analyzer (since it has a more generic name) implement the logic of CA1051 as well.